src/components/CreateGameServer/CreateGameServerModal.tsx

```diff
@@ -0,0 +1,95 @@
+import { Button } from "@components/ui/button.tsx";
+import { DialogContent, DialogFooter } from "@components/ui/dialog.tsx";
+import { createContext, useCallback, useState } from "react";
+import type { GameServerCreationDto } from "@/api/generated/model";
+import useDataInteractions from "@/hooks/useDataInteractions/useDataInteractions";
+import useTranslationPrefix from "@/hooks/useTranslationPrefix/useTranslationPrefix";
+import Step1 from "./CreationSteps/Step1";
+import Step2 from "./CreationSteps/Step2";
+import Step3 from "./CreationSteps/Step3";
+
+export interface GameServerCreationContext {
+  gameServerState: Partial<GameServerCreationDto>;
+  setGameServerState: (
+    gameStateKey: keyof GameServerCreationDto,
+  ) => (value: GameServerCreationDto[keyof GameServerCreationDto]) => void;
```

Comment:
Improved typing through this:
```suggestion
  setGameServerState: <K extends keyof GameServerCreationDto>(
  gameStateKey: K
) => (value: GameServerCreationDto[K]) => void;
```